spec/rspec/rails/matchers/be_a_new_spec.rb

```diff
@@ -79,12 +79,7 @@ def new_record?; true; end
 
           context "matcher is wrong type" do
             it "fails" do
-              expect {
-                expect(record).to be_a_new(record.class).with(
-                  foo: a_hash_including({ no_foo: "foo" }))
```

Comment:
What if we just use a different "wrong type" of matcher here and keep the example as is? Like `be_within`/`cover`/`contain_exactly`.
At the same time, this example won't break after your rspec-expectations fix.